pkg/cmd/cli/cli_changefeed_resume.go

```diff
@@ -62,44 +74,92 @@ func (o *resumeChangefeedOptions) complete(f factory.Factory) error {
 	return nil
 }
 
+func (o *resumeChangefeedOptions) getTSO(ctx context.Context) (*v2.Tso, error) {
+	tso, err := o.apiV2Client.Tso().Query(ctx,
+		&v2.UpstreamConfig{ID: o.changefeedDetail.UpstreamID})
+	if err != nil {
+		return nil, err
+	}
+
+	return tso, nil
+}
+
+func (o *resumeChangefeedOptions) getChangefeedInfo(ctx context.Context) (
+	*model.ChangefeedDetail, error,
+) {
+	detail, err := o.apiV1Client.Changefeeds().Get(ctx, o.changefeedID)
+	if err != nil {
+		return nil, err
+	}
+
+	return detail, nil
+}
+
 // confirmResumeChangefeedCheck prompts the user to confirm the use of a large data gap when noConfirm is turned off.
 func (o *resumeChangefeedOptions) confirmResumeChangefeedCheck(ctx context.Context, cmd *cobra.Command) error {
 	if !o.noConfirm {
-		cfs, err := o.apiV1Client.Changefeeds().List(ctx, "all")
-		if err != nil {
-			return err
+		if len(o.overwriteCheckpointTs) == 0 {
+			return confirmLargeDataGap(cmd, o.currentTso.Timestamp,
+				o.changefeedDetail.CheckpointTSO)
 		}
-		var checkpointTs uint64 = 0
-		var upstreamID uint64 = 0
 
-		for _, cf := range *cfs {
-			if cf.ID == o.changefeedID {
-				checkpointTs = cf.CheckpointTSO
-				upstreamID = cf.UpstreamID
-			}
-		}
-		if checkpointTs == 0 {
-			return errors.ErrChangeFeedNotExists
-		}
+		return confirmOverwriteCheckpointTs(cmd, o.changefeedID, o.checkpointTs)
+	}
+	return nil
+}
 
-		tso, err := o.apiV2Client.Tso().Query(ctx,
-			&v2.UpstreamConfig{ID: upstreamID})
-		if err != nil {
-			return err
-		}
-		return confirmLargeDataGap(cmd, tso.Timestamp, checkpointTs)
+func (o *resumeChangefeedOptions) validateParams(ctx context.Context, cmd *cobra.Command) error {
+	// check whether the changefeed to be resumed is existing
+	detail, err := o.getChangefeedInfo(ctx)
+	if err != nil {
+		return err
+	}
+	o.changefeedDetail = detail
+
+	tso, err := o.getTSO(ctx)
+	if err != nil {
+		fmt.Println("error occurs when getTSO")
+		return err
+	}
+	o.currentTso = tso
+
+	if len(o.overwriteCheckpointTs) == 0 {
+		return nil
+	}
+
+	// validate the --overwrite-checkpoint-ts parameter
+	if strings.ToLower(o.overwriteCheckpointTs) == "now" {
+		o.checkpointTs = oracle.ComposeTS(tso.Timestamp, tso.LogicTime)
+		return nil
 	}
+
+	checkpointTs, err := strconv.ParseUint(o.overwriteCheckpointTs, 10, 64)
+	if err != nil {
+		return errors.New("invalid overwrite-checkpoint-ts, " +
```

Comment:
Do we need an RFC code for these errors?

Reply:
> Do we need an RFC code for these errors?

It's reasonable to add RFC codes for these errors. I'll add it later.